### PR TITLE
Adds documentation for Dell EMC ScaleIO volume plugin

### PR DESCRIPTION
**Pull request by vladimirvivien**
This commit updates the volumes documentation to include ScaleIO volume plugin information.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2741)

### CODE REVIEW

File: docs/user-guide/persistent-volumes/index.md [hunk 1]
```diff
@@ -534,6 +535,33 @@ parameters:
...
+  fsType: xfs
+```
+
+* `provisioner`: attribute is set to `kubernetes.io/scaleio`
+* `gateway`: address to a ScaleIO API gateway
+* `system`: the name of the ScaleIO system
+* `protectionDomain`: the name of the ScaleIO protection domain
+* `storagePool`: the name of the volume storage pool
+* `storageMode`: the way storage is provisioned (ThinProvisionned or ThickProvisionned)
+* `secretRef`: name reference to a configuered Secret object
```
Comment (jsafrane) [inline docs/user-guide/persistent-volumes/index.md]: * In what namespace is the secret expected?
* What are expected keys and values of the secret? Please follow example e.g. from quobyte above.

File: docs/user-guide/persistent-volumes/index.md [hunk 2]
```diff
@@ -534,6 +535,33 @@ parameters:
 * `location`: Azure storage account location. Default is empty.
 * `storageAccount`: Azure storage account name. If storage account is not provided, all storage accounts associated with the resource group are searched to find one that matches `skuName` and `location`. If storage account is provided, it must reside in the same resource group as the cluster, and `skuName` and `location` are ignored.
 
+#### ScaleIO
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
```
Comment (jsafrane) [inline docs/user-guide/persistent-volumes/index.md]: Please use `storage.k8s.io/v1`, I'm waiting for https://github.com/kubernetes/kubernetes.github.io/pull/2686 to get merged.

File: docs/user-guide/persistent-volumes/index.md (hunk 1 above)
Comment (vladimirvivien) [inline docs/user-guide/persistent-volumes/index.md]: @jsafrane Hi, I updated the doc to address your comments about namespace values.

File: docs/user-guide/persistent-volumes/index.md (hunk 2 above)
Comment (vladimirvivien) [inline docs/user-guide/persistent-volumes/index.md]: Done!

**Comment by jsafrane** [comment]
Thanks for the PR.
LGTM from tech side.

**Comment by vladimirvivien** [comment]
@jsafrane thank for the review.

**Comment by vladimirvivien** [comment]
@jsafrane is there anything else I need to do for this to get merged?  Thanks.

**Comment by jsafrane** [comment]
Someone from docs team should merge it, I do just technical review. To be honest, I don't understand why it's assigned to me. @chenopis, PTAL.

**Pull request merged**